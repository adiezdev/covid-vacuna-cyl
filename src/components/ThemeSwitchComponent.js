import React from 'react'

export const ThemeSwitchComponent = () => {

    return (
        <div className='switch'>
            <div
                className='themeLight'
                role='img'
            >
            🌞
            </div>
            <div
                className='themeSystme'
                role='img'
            >
            💻
            </div>
            <div
                className='themeDark'
                role='img'
            >
            🌚
            </div>
        </div>
    )
}
