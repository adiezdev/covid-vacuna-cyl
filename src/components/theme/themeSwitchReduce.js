

export const themeSwitchReduce = ( state = [],action )=>{

    switch (action.type) {
        case 'light':
            
             document.documentElement.setAttribute('data-theme',action.type)
             localStorage.setItem('data-theme',action.type)
             return 'isActivated';

         case 'dark':

             document.documentElement.setAttribute('data-theme',action.type)
             localStorage.setItem('data-theme',action.type)
             return 'isActivated';

        default:
            break;
    }
}