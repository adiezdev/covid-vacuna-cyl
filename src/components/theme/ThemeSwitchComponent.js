import React, { useEffect, useReducer } from 'react'
import { useLocalStorage } from '../../hooks/useLocalStorage';
import { themeSwitchReduce } from './themeSwitchReduce';

/**
 * Check init localstorage
*/

export const ThemeSwitchComponent = () => {

    const [state, dispatch] = useReducer(themeSwitchReduce , '')
    //Inicializamos con el theme del sistema
    const [storagevalues , setStorage] = useLocalStorage('data-theme','other')

    /**
     * Lee el valor por guardado del hook local storage
     */
    useEffect(() => {
        //cambia el tema en el sistema por el que esté
        dispatch({
            type: storagevalues
        })

    }, [storagevalues])

    //Llamada para poder cambiar manualmente el theme de la pagina
    const  handleOnClick = ( type )  => {
        //disparador
        dispatch({ type: type })
        //cambio en el localstorage
        setStorage(type)
    }
    /**
     * If useReducer return state ative, check is same to type and add class isActiviated
     */
    return (
        <div className='switchComponent'>
            <label
                className={`theme ${state.type === 'light' ? state.active : ''}`}
                onClick={() => handleOnClick('light')}
            >
                <span role='img'>☀️</span>
            </label>
            <label
                 className={`theme ${state.type === 'other' ? state.active : ''}`}
                 onClick={() => handleOnClick('other')}

            >
                <span role='img'>💻</span>
            </label>
            <label
                 className={`theme ${state.type === 'dark' ? state.active : ''}`}
                 onClick={() => handleOnClick('dark')}

            >
                <span role='img'>🌑</span>
            </label>
        </div>
    )
}
