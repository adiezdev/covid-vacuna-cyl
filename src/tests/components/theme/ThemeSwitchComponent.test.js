import Enzyme, { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'

import { ThemeSwitchComponent } from '../../../components/theme/ThemeSwitchComponent'


import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });



describe('Pruebas al componente <TheSwitchComponent/>', () =>{

 

    test('Debe de mostrarse correctamente', () =>{

        const wrapper = shallow(<ThemeSwitchComponent/>)

        expect(shallowToJson(wrapper)).toMatchSnapshot()
    })

})