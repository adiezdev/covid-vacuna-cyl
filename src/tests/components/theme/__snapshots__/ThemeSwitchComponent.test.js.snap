// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <TheSwitchComponent/> Debe de mostrarse correctamente 1`] = `
<div
  className="switchComponent"
>
  <label
    className="theme "
    onClick={[Function]}
  >
    <span
      role="img"
    >
      ☀️
    </span>
  </label>
  <label
    className="theme "
    onClick={[Function]}
  >
    <span
      role="img"
    >
      💻
    </span>
  </label>
  <label
    className="theme "
    onClick={[Function]}
  >
    <span
      role="img"
    >
      🌑
    </span>
  </label>
</div>
`;
